src/assets/wise5/directives/componentAnnotations/component-annotations.component.ts

```diff
@@ -45,34 +57,88 @@ export class ComponentAnnotationsComponent {
         }
       }
     );
+    this.wiseLinkCommunicatorId = `wise-link-communicator-component-annotations-${this.nodeId}-${this.componentId}`;
   }
 
-  ngOnChanges() {
+  ngAfterViewInit() {
     this.processAnnotations();
+    this.wiseLinkCommunicator = document.getElementById(this.wiseLinkCommunicatorId);
+    if (this.wiseLinkCommunicator != null) {
+      this.wiseLinkClickedHandler = this.WiseLinkService.createWiseLinkClickedHandler(this.nodeId);
+      this.WiseLinkService.addWiseLinkClickedListener(
+        this.wiseLinkCommunicator,
+        this.wiseLinkClickedHandler
+      );
+    }
+  }
+
+  ngOnChanges(changes: SimpleChanges) {
+    if (!changes.annotations.isFirstChange()) {
+      this.processAnnotations();
+    }
   }
 
   ngOnDestroy() {
     this.studentWorkSavedToServerSubscription.unsubscribe();
+    if (this.wiseLinkClickedHandler != null) {
+      this.wiseLinkCommunicator.removeEventListener('wiselinkclicked', this.wiseLinkClickedHandler);
+    }
   }
 
-  processAnnotations() {
+  processAnnotations(): void {
     if (this.annotations.comment || this.annotations.score) {
-      this.nodeId = this.annotations.comment
-        ? this.annotations.comment.nodeId
-        : this.annotations.score.nodeId;
-      this.componentId = this.annotations.comment
-        ? this.annotations.comment.componentId
-        : this.annotations.score.nodeId;
-      this.showScore =
-        this.annotations.score != null &&
-        this.ProjectService.displayAnnotation(this.annotations.score);
-      this.showComment =
-        this.annotations.comment != null &&
-        this.ProjectService.displayAnnotation(this.annotations.comment);
+      this.nodeId = this.getNodeId(this.annotations);
+      this.componentId = this.getComponentId(this.annotations);
+      this.showScore = this.isShowScore(this.annotations);
+      this.showComment = this.isShowComment(this.annotations);
+      if (this.showComment) {
+        this.comment = this.getCommentHtml(this.annotations.comment);
+      }
       this.setLabelAndIcon();
     }
   }
 
+  getNodeId(annotations: any): string {
+    return this.hasCommentAnnotation(annotations)
+      ? annotations.comment.nodeId
+      : annotations.score.nodeId;
+  }
+
+  getComponentId(annotations: any): string {
+    return this.hasCommentAnnotation(annotations)
+      ? annotations.comment.componentId
+      : annotations.score.nodeId;
```

Comment:
Should this be ```annotations.score.componentId;``` ? 

I see that it is nodeId in the original code.

Reply:
Ah, I missed that. I've fixed it now.